src/test_l1_dsFPD.c

```diff
@@ -744,86 +727,86 @@ void test_l1_dsFPD_negative_dsSetFPBrightness (void)
+    count = UT_KVP_PROFILE_GET_UINT8("dsFPD/Number_of_Indicators");
     for(int i = 1; i <= count; i++)
     {
 	snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
-        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        eIndicator =(dsFPDIndicator_t)UT_KVP_PROFILE_GET_UINT32(buffer);
         snprintf(minbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/MIN_BRIGHTNESS",eIndicator);
-        minBrightness =  UT_KVP_PROFILE_GET_UINT32(minbuffer);
+        minBrightness =(dsFPDBrightness_t)UT_KVP_PROFILE_GET_UINT32(minbuffer);
         snprintf(maxbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/MAX_BRIGHTNESS",eIndicator);
-        maxBrightness = UT_KVP_PROFILE_GET_UINT32(maxbuffer);
-        int avgBrightness = (maxBrightness + minBrightness) / 2;
+        maxBrightness =(dsFPDBrightness_t)UT_KVP_PROFILE_GET_UINT32(maxbuffer);
+        uint32_t avgBrightness = (maxBrightness + minBrightness) / 2;
         result = dsSetFPBrightness(eIndicator, avgBrightness);
         CHECK_FOR_EXTENDED_ERROR_CODE( result, dsERR_NOT_INITIALIZED, dsERR_NONE);
     }
 
-    // Variation 02: Initialize with dsFPInit()
+    // Step 02: Initialize with dsFPInit()
     result = dsFPInit();
     UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
-    // Variation 03: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
-    // Variation 04: Pass an invalid eBrightness parameter to dsSetFPBrightness()
+    // Step 03: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
+    // Step 04: Pass an invalid eBrightness parameter to dsSetFPBrightness()
     for(int i = 1; i <= count; i++)
     {
 	snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
-        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        eIndicator =(dsFPDIndicator_t)UT_KVP_PROFILE_GET_UINT32(buffer);
         UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, eIndicator);
         result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
         result = dsSetFPBrightness(eIndicator, 200);
         UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
     }
 
-    // Variation 05: Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()
+    // Step 05: Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()
     for(int i = 1; i <= count; i++)
     {
         snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
-        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        eIndicator =(dsFPDIndicator_t)UT_KVP_PROFILE_GET_UINT32(buffer);
         result = dsSetFPState(eIndicator, dsFPD_STATE_OFF);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
     }
 
-    // Variation 06: Call dsSetFPBrightness() and loop through all valid indicators from yaml
+    // Step 06: Call dsSetFPBrightness() and loop through all valid indicators from yaml
```

Comment:
Line 777 to 794 looks more of a positive kind...the scenario is to test by setting the brightness when the LED State is set to OFF.  We need to remove the LED State setting to ON and try to set the brightness which should fail.  Recheck these steps fully.

Reply:
When you set off, the API should return a valid error code, and also no need to set the LED to off again at the end as its already made OFF.

Replace:
        result = dsSetFPBrightness(eIndicator, avgBrightness);
        UT_ASSERT_EQUAL(result, dsERR_NONE);
        result = dsSetFPState(eIndicator, dsFPD_STATE_OFF);
        UT_ASSERT_EQUAL(result, dsERR_NONE);

To 
        result = dsSetFPBrightness(eIndicator, avgBrightness);
        UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);

Reply:
Done